internal/commands/user/inputs.go

```diff
@@ -0,0 +1,120 @@
+package user
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+const (
+	userDataEmail = "email"
+	userDataName  = "name"
+)
+
+// usersInputs are the filtering inputs for a user command
+type usersInputs struct {
+	State         realm.UserState
+	ProviderTypes []string
+	Status        statusType
+	Users         []string
+}
+
+// ResolveUsers will use the provided Realm client to resolve the users specified by the realm.App through inputs
+func (i *usersInputs) ResolveUsers(ui terminal.UI, client realm.Client, app realm.App) ([]realm.User, error) {
+	if len(i.Users) > 0 {
+		filter := realm.UserFilter{IDs: i.Users}
+		foundUsers, err := client.FindUsers(app.GroupID, app.ID, filter)
+		if len(foundUsers) == 0 || err != nil {
+			return nil, err
+		}
+		if len(foundUsers) == 0 {
+			return nil, errors.New("No users found from provided IDs")
+		}
+		return foundUsers, nil
+	}
+
+	userFilter := realm.UserFilter{}
+	if len(i.ProviderTypes) > 0 {
+		userFilter.Providers = i.ProviderTypes
+	}
+	if i.State != realm.UserStateNil {
+		userFilter.State = i.State
+	}
+	selectableUsersSet := map[string]realm.User{}
+	if i.Status == statusTypeConfirmed || i.Status == statusTypeEmpty {
+		foundUsers, err := client.FindUsers(app.GroupID, app.ID, userFilter)
+		if err != nil {
+			return nil, err
+		}
+		for _, user := range foundUsers {
+			selectableUsersSet[user.ID] = user
+		}
+	}
+	if i.Status == statusTypePending || i.Status == statusTypeEmpty {
+		userFilter.Pending = true
+		foundUsers, err := client.FindUsers(app.GroupID, app.ID, userFilter)
+		if err != nil {
+			return nil, err
+		}
+		for _, user := range foundUsers {
+			selectableUsersSet[user.ID] = user
+		}
+	}
+
+	// Interactive User Selection
+	selectableUsers := map[string]realm.User{}
+	selectableUserOptions := make([]string, len(selectableUsersSet))
```

Comment:
same thought here about initializing this to have 0 length but the right capacity, and use `append` below